frontend/app/routes/$lang+/_protected+/personal-information+/mailing-address+/edit.tsx

```diff
@@ -143,14 +150,23 @@ export async function action({ context: { session }, params, request }: ActionFu
 
   instrumentationService.countHttpStatus('mailing-address.edit', 302);
 
-  //TODO: need updatePersonalInfo to update home address
-  //await addressService.updateAddressInfo(userId, userInfo?.homeAddress ?? '', newHomeAddress);
-  session.set('newMailingAddress', parsedDataResult.data);
+  const updatedAddress = {
+    address: parsedDataResult.data.streetName,
+    city: parsedDataResult.data.cityName,
+    province: parsedDataResult.data.provinceTerritoryStateId,
+    postalCode: parsedDataResult.data.postalCode,
+    country: parsedDataResult.data.countryId,
+  }
+  // TODO: secondAddressLine is never used and never saved.
+  await getAddressService().updateAddressInfo(userId, userInfo?.mailingAddress ?? '', updatedAddress);
+  if (parsedDataResult.data.homeAndMailingAddressTheSame) { 
+    await getAddressService().updateAddressInfo(userId, userInfo?.homeAddress ?? '', updatedAddress);
+  }
```

Comment:
I would leave the previous code and corresponding `TODO` as there is currently a user story (AB#3233) to create a (mock) service operation to save personal information.

The plan is to eventually remove `address-service` and `user-service` in favour of the `personal-information-service`, so lets try to leave the references to the first two services out for now.